bpfman/src/lib.rs

```diff
@@ -93,7 +93,116 @@ pub(crate) fn get_db_config() -> SledConfig {
     SledConfig::default().temporary(true)
 }
 
-/// Loads an ebpf program.
+/// Adds an eBPF program to the system.
+///
+/// This function takes a `Program` and performs the necessary loading
+/// and attaching operations to add it to the system. It supports
+/// various types of eBPF programs such as XDP, TC, Tracepoint,
+/// Kprobe, Uprobe, Fentry, and Fexit. The program can be added from a
+/// locally built bytecode file or a remote bytecode image. If the
+/// program is successfully added, it returns the updated [`Program`];
+/// otherwise, it returns a `BpfmanError`.
+///
+/// # Arguments
+///
+/// * `program` - The eBPF program to be added.
+///
+/// # Returns
+///
+/// * `Ok(Program)` - If the program is successfully added.
+/// * `Err(BpfmanError)` - If there is an error during the process.
+///
+/// # Example
+///
+/// ```rust,no_run
+/// use bpfman::add_program;
+/// use bpfman::errors::BpfmanError;
+/// use bpfman::types::{KprobeProgram, Location, Program, ProgramData};
+/// use std::collections::HashMap;
+///
+/// #[tokio::main]
+/// async fn main() -> Result<(), BpfmanError> {
+///     // Define the location of the eBPF object file.
+///     let location = Location::File(String::from("kprobe.o"));
+///
+///     // Initialise metadata for the program. Metadata contains custom
+///     // configurations or properties related to the eBPF program, such
+///     // as threshold values, sampling rates, specific flags (e.g.,
+///     // enabling debug mode), and other parameters. These values can be
+///     // used by the userspace program.
+///     let mut metadata = HashMap::new();
+///     metadata.insert("cpu_usage_threshold".to_string(), "80".to_string());
+///     metadata.insert("sampling_rate".to_string(), "1000".to_string());
+///     metadata.insert("enable_debug".to_string(), "true".to_string());
```

Comment:
Is this chunk of code needed (the metadata.insert() calls)? I'm not sure what it's purpose is? Plus, the description is wrong, the correct text is below.

Reply:
The insert calls were just examples. I can leave it at: 

```rust
let mut metadata = HashMap::new();
```

if you think that's sufficient.